### ISSUE TITLE
Dependency conflict

### ISSUE THREAD

**Issue by macbef** [opencv-pg/opencv-pg#16]
Hello,
Does anybody know how to avoid this conflict when running "pip install opencv-pg"?

ERROR: Cannot install opencv-pg==1.0.0 and opencv-pg==1.0.1 because these package versions have conflicting dependencies.
The conflict is caused by:
    opencv-pg 1.0.1 depends on pyside2==5.15.0
    opencv-pg 1.0.0 depends on pyside2==5.15.0

Thanks for your help!

**Comment by geoff-va** [comment]
Was this happening from a fresh virtual environment?

**Comment by macbef** [comment]
pretty much the same.

I'm on a Windows 10 machine. Python 3.10.11 / 32bit 
In a new directory I ran:

`python -m venv .`

`pip install opencv-pg`

I get this:

```
Collecting opencv-pg
  Using cached opencv_pg-1.0.1-py3-none-any.whl (147 kB)
INFO: pip is looking at multiple versions of opencv-pg to determine which version is compatible with other requirements. This could take a while.
  Using cached opencv_pg-1.0.0-py3-none-any.whl (147 kB)
ERROR: Cannot install opencv-pg==1.0.0 and opencv-pg==1.0.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    opencv-pg 1.0.1 depends on pyside2==5.15.0
    opencv-pg 1.0.0 depends on pyside2==5.15.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Comment by geoff-va** [comment]
Did you first activate the newly created virtualenv before the pip install step?

**Comment by geoff-va** [comment]
@macbef were you able to resolve this?

**Comment by jeonghyunjo** [comment]
Having same issue here. Was it solved? I'm on Ubuntu 18.04.

**Comment by geoff-va** [comment]
Can you please list the exact steps you took to install it all?

**Comment by jeonghyunjo** [comment]
Simply I ran "pip install opencv-pg" and I got the same error which is 
`The conflict is caused by:
    opencv-pg 1.0.1 depends on pyside2==5.15.0
    opencv-pg 1.0.0 depends on pyside2==5.15.0`

**Comment by geoff-va** [comment]
I just tried that and saw the same message - I have no idea why that's happening... none of the requirements would seem to indicate it should do this. Seems something about whatever pypi is doing?

Try `pip install opencv-pg==1.0.1` specifically in a clean virtualenv.

**Comment by jeonghyunjo** [comment]
It worked! I'll test with this version for now.

**Comment by jeonghyunjo** [comment]
Loading image doesn't work.

**Comment by geoff-va** [comment]
Great! Hopefully that gets you going.

I’ll see if I can dig into what’s going on with the pypi install. Seems to
be wanting to install both versions for some reason when not a specific one
is specified. Very odd.

**Comment by geoff-va** [comment]
Does the test image work? Or what happens?

**Comment by jeonghyunjo** [comment]
When I use this command, image doesn't show up on the gui
`opencvpg --image <path-to-image.png>`

**Comment by geoff-va** [comment]
Ah but the test image works?

**Comment by jeonghyunjo** [comment]
Any image doesn't show up on the screen. 
![image](https://github.com/opencv-pg/opencv-pg/assets/77941513/fbf03943-e2e7-44f5-b2d9-b9827cd04f9a)

**Comment by geoff-va** [comment]
Just to confirm nothing shows up when you run opencvpg alone either?

Unfortunately I don’t have hardware that can build this version anymore. I need to find a suitable set of deps to get it to build/install before I could troubleshoot anything.

**Comment by jeonghyunjo** [comment]
Yes nothing shows up with and without image. I see. Thank you and I'll be waiting for other answers too.

**Comment by geoff-va** [comment]
One other question - do you see any output in the console by any chance after it loads?

**Comment by jeonghyunjo** [comment]
No printout after I run command. Some traceback shows up when I click the functions but I don't think it's related to the image.

**Comment by geoff-va** [comment]
Can you please paste those tracebacks and what action caused it?

**Comment by geoff-va** [comment]
@jeonghyunjo If you're interested in being a linux test-bed - I was able to get it running on my system after doing some package upgrades and fixes when going from from Qt5 -> Qt6.

If you're interested - you could install the `1.1.0` version on the `feat/upgrade-to-qt6` branch:

```shell
git clone https://github.com/opencv-pg/opencv-pg
cd opencv-pg
git checkout feat/upgrade-to-qt6
python -m venv ./.venv-qt6
source ./.venv-qt6/bin/activate
pip install ./
```

**NOTE:** The webview is still broken and I haven't looked at how to fix it yet (at least on MacOs it is). I've been able to run it with: `opencvpg --no-docs`

**Comment by geoff-va** [comment]
Hey @jeonghyunjo I was able to get all the core functionality working again on my system (save some mathjax rendering in the webview apparently). Will look at that next.

Let me know if you happen to get a chance to try and install it in a clean venv and whether or not it's working.

I'm using an M1 mac and it's worked on python 3.8.10 and 3.10.0.

Thanks!

**Comment by geoff-va** [comment]
@macbef I've release a 1.0.2 version and was able to successfully install it via a normal `pip install opencv-pg`. I've also upgraded to Qt6 (I couldn't build the original version on an M1 mac and no longer have access to any other kind of device). 

If still trouble you could try the `--no-cache` option, also.

Hope that helps -
